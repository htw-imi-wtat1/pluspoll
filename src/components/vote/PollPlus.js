import React from "react";

import Option from './Option'
import NewOption from "../shared/NewOption";
import label2Option from "../shared/label2Option";

import socketIOClient from "socket.io-client";

const ENDPOINT = "http://127.0.0.1:3001";

class PollPlus extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            question: props.poll.question,
            options: props.poll.options,
            votes: props.poll.votes
        };
        //this.optionChecked = this.optionChecked.bind(this);
        this.createHandler = this.createHandler.bind(this)
        this.addOption = this.addOption.bind(this)
    }
    componentDidMount(){
        const socket  = socketIOClient(ENDPOINT);
        socket.on("addOption", data => {
            console.log("new option received: "+JSON.stringify(data))
            const options = this.state.options
            options.push(data)
            console.log("options "+JSON.stringify(options))
            this.setState({options: options})

        });
        this.socket = socket
    }
    createHandler(name){
        const optionChecked = (event) => {
            const checked = event.target.checked
            const currentVotes = this.state.votes
            currentVotes[name] = checked
            console.log("option checked: "+name+checked)
            console.log(JSON.stringify(currentVotes))
            this.setState({votes: currentVotes})
        }
        return optionChecked
    }
    addOption(newOption){
        console.log("add option called: "+newOption)
        this.socket.emit("addOption",label2Option(newOption))
    }

    render() {
        console.log("re-render")
        // console.log(JSON.stringify(this.state))

        return (
            <div>
                <h2>{this.state.question}</h2>
                {this.state.options.map((option) => {
                    const votes = this.state.votes
                    return (<Option key = {option.name} name ={option.name} label = {option.label} checked={votes[option.name]} onChange = {this.createHandler(option.name)}/>)
                })}
                <NewOption key="newOption" onChange = {this.addOption}/>

            </div>
        )
    }
}
export default PollPlus