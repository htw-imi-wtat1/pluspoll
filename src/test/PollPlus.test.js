import React from 'react';
import snapshotRenderer from 'react-test-renderer'
import PollPlus from '../components/vote/PollPlus';


const poll =
    {
        question: 'Please choose an icecream flavor!',
        options: [{name: "rumraisin", label: "Rum-Raisin"},
            {name: "vanilla", label: "Vanilla"},
            {name: "chocolate", label: "Chocolate"},
            {name: "pistachio", label: "Pistachio"}],
        votes: {
            vanilla: false,
            chocolate: false,
            pistachio: false,
            rumraisin: false
        }
    }
test('renders poll creator', () => {
    const component = snapshotRenderer.create(
        <PollPlus poll =  {poll} />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    const newQuestion = "What's your favourite programming language?"
});
