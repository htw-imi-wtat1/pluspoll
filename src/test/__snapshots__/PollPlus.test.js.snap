// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders poll creator 1`] = `
<div>
  <h2>
    Please choose an icecream flavor!
  </h2>
  <div>
    <label>
      <input
        checked={false}
        name="rumraisin"
        onChange={[Function]}
        type="checkbox"
      />
      Rum-Raisin
    </label>
  </div>
  <div>
    <label>
      <input
        checked={false}
        name="vanilla"
        onChange={[Function]}
        type="checkbox"
      />
      Vanilla
    </label>
  </div>
  <div>
    <label>
      <input
        checked={false}
        name="chocolate"
        onChange={[Function]}
        type="checkbox"
      />
      Chocolate
    </label>
  </div>
  <div>
    <label>
      <input
        checked={false}
        name="pistachio"
        onChange={[Function]}
        type="checkbox"
      />
      Pistachio
    </label>
  </div>
  <div>
    <label>
      + 
      <input
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="add option"
        value=""
      />
    </label>
  </div>
</div>
`;
